### ISSUE TITLE
Breaking change in SteamVR beta: Expected overlay intersection coordinates changed

### ISSUE THREAD

**Issue by bddckr** [ValveSoftware/openvr#917]
On the latest SteamVR betas (last tested on 1539932350) the interaction with (dashboard or regular) overlays is broken:

If an overlay uses `SetOverlayIntersectionMask` the laser pointer is now actually completely broken with regards to aiming at the overlay. What worked previously (and does on the non-beta branch) now results in both stretched collision areas as well as wholes in it.

This may be affecting overlays that are moved, rotated or changed in size only as it seems to be fine for SteamVR's own dashboard overlays, but using the API to change your own overlay in any such way results in this broken pointing behavior.

*Edit:* Can confirm it's also happening to regular, non-dashboard overlays.

**Comment by bddckr** [comment]
The beta update 1539997838 did fix it for overlays that aren't transformed, i.e. regular sized, positioned and rotated overlays. **The issue still affects any other overlay.**

There seems to be no improvement to those overlays with that update.

**Comment by bddckr** [comment]
Upon further inspection it seems like this is happening when using `SetOverlayIntersectionMask` **and** `SetOverlayTextureBounds`.

If this is a breaking change we should be handling I'd love to hear how the coordinate system works for this as I've been doing some math to adjust the input I pass to `SetOverlayIntersectionMask` that has to scale the input by the texture's size to make it usable. Said math still works fine on the x-axis but no longer does on the y-axis:
The laser pointer keeps "hitting" the overlay in areas as if the whole collision rectangle is stretched to double its height or something.

**Comment by dag10** [comment]
This shouldn't be a breaking change on our end. Do you have a repro we can see?

**Comment by bddckr** [comment]
Thanks a lot @dag10!

I am in the process of preparing a repro project now and will ping you on this in here ASAP.

**Comment by bddckr** [comment]
Sorry for this taking some time. It took some time to notice that we can now just specify the mouse scale as well as any intersection coordinates in absolute coordinates relative to the whole texture, despite just drawing a part of it via the texture bounds.

While creating a stripped down version to make reproducing the issue easier we found that the betas actually **come with a breaking change, but that change is for the better**:
* Previously (i.e. the current regular version of SteamVR) the *intersection mask API* was directly bound to the texture bounds and also had its origin at the *bottom left*.
* Now (i.e. in the current betas) the same API is independent from the texture bounds and its origin is at the *top left*.

These two changes, despite being breaking changes, make the API easier to use as well as prevent the need to find out about these details in the first place. I would classify these changes as bugfixes actually as the previous behavior was confusing and unexpected.

Note that the input events you get by polling the overlay events are still using the *bottom left* as the origin - this didn't change at all and is fine.

---

**tl;dr**: All good! The betas come with a breaking change, but that change makes the API easier!

Thanks again @dag10!
